modules/basegl/src/processors/raycasting/texturedisosurfacerenderer.cpp

```diff
@@ -0,0 +1,280 @@
+            result = drawISO(result, isoparams.values[i], isoparams.colors[i].a, value, previousValue,
+                             gradient, previousGradient, textureToWorld, lighting, rayPosition,
+                             rayDirection, toCameraDir, t, tIncr, tDepth);
+        }
+    } else {
+        for (int i = isoparams.size; i > 0; --i) {
+            result = drawISO(result, isoparams.values[i - 1], isoparams.colors[i - 1].a, value,
+                             previousValue, gradient, previousGradient, textureToWorld, lighting,
+                             rayPosition, rayDirection, toCameraDir, t, tIncr, tDepth);
+        }
+    }
+    #endif
+
+    return result;
+}
+
+)");
+
+constexpr std::string_view classify = util::trim(R"(
+result = drawISO(result, {iso}, {volume}Voxel[channel], {volume}VoxelPrev[channel],
+                 {volume}Gradient, {volume}GradientPrev, {volume}Parameters.textureToWorld,
+                 lighting, samplePosition, rayDirection,
+                 cameraDir, rayPosition, rayStep, rayDepth);
+)");
+
+constexpr std::string_view uniforms = util::trim(R"(
+uniform VolumeParameters {0}Parameters;
+uniform sampler3D {0};
+)");
+
+}  // namespace
+
+auto TexturedIsoSurfaceComponent::getSegments() -> std::vector<Segment> {
+    using namespace fmt::literals;
+
+    std::vector<Segment> segments{
+        {fmt::format(FMT_STRING(uniforms), colorPort.getIdentifier()), placeholder::uniform, 400},
+
+        {std::string(R"(#include "utils/compositing.glsl")"), placeholder::include, 1100},
+        {std::string(R"(uniform int channel = 0;)"), placeholder::uniform, 1100},
+        {std::string(R"(uniform int colorChannel = 0;)"), placeholder::uniform, 1101},
+
+        {fmt::format(isoCalc, "color"_a = colorPort.getIdentifier(), "tf"_a = tf->getIdentifier()),
+         placeholder::uniform, 3000},
+        {std::string{isoDraw}, placeholder::uniform, 3005},
+
+        {fmt::format(classify, "color"_a = colorPort.getIdentifier(), "volume"_a = volume,
+                     "tf"_a = tf->getIdentifier(), "iso"_a = iso->getIdentifier()),
+         placeholder::first, 600},
+        {fmt::format(classify, "color"_a = colorPort.getIdentifier(), "volume"_a = volume,
+                     "tf"_a = tf->getIdentifier(), "iso"_a = iso->getIdentifier()),
+         placeholder::loop, 600},
+    };
+
+    return segments;
+}
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo TexturedIsoSurfaceRenderer::processorInfo_{
+    "org.inviwo.TexturedIsoSurfaceRenderer",  // Class identifier
+    "Textured Iso Surface Renderer",          // Display name
+    "Volume Rendering",                       // Category
+    CodeState::Stable,                        // Code state
+    Tags::GL,                                 // Tags
+    R"(Render a iso surface defined by the Volume data.
+       The iso surface is textured by the second color
```

Comment:
The second volume is not a color volume? 
If that were so there would be no need for a transfer function (we could just use the color volume).

The iso surface retrieves its color using the second volume’s value mapped through the transfer function. This makes it possible to visualize the other volume’s value at the the iso surface location

Reply:
No the second volume is not a color volume

Reply:
Ok, then I suggest that we update the text to something in the line of:
Render an iso surface defined by the Volume data, but color it based on the values of a second volume.
The iso surface retrieves its color using the second volume’s value mapped through the transfer function. This makes it possible to visualize the other volume’s value at the the iso surface location.

Then we are good to go